### PR TITLE
disttask,external: implement merge step

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #45719 close https://github.com/pingcap/tidb/issues/47011

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/backfilling_dispatcher.go [hunk 1]
```diff
@@ -114,14 +106,41 @@ func (h *backfillingDispatcherExt) OnNextSubtasksBatch(ctx context.Context,
...
 		return proto.StepOne
 	case proto.StepOne:
+		// TODO(lance6716): temporary force to use merge step
 		return proto.StepTwo
+
+		// if data files overlaps too much, we need a merge step.
+		subTaskMetas, err := taskHandle.GetPreviousSubtaskMetas(task.ID, proto.StepInit)
+		if err != nil {
+			// TODO(lance6716): should we return error?
+			return proto.StepTwo
```
Comment (wjhuang2016) [inline ddl/backfilling_dispatcher.go:124]: Return the error and retry if it's retriable.

File: ddl/backfilling_merge_sort.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+import (
+	"context"
+	"encoding/json"
+	"path"
+	"strconv"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/pingcap/tidb/br/pkg/storage"
```
Comment (wjhuang2016) [inline ddl/backfilling_merge_sort.go:25]: lint

File: ddl/backfilling_dispatcher.go (hunk 1 above)
Comment (ywqzzy) [reply ddl/backfilling_dispatcher.go:124]: I prefer to do it in next pr, this may introduce some conflicts.

Comment (lance6716) [reply ddl/backfilling_dispatcher.go:121]: I think so, we can merge this PR ASAP

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**